Add several replacements to escapeXML.pl script

chbrown opened the pull request:
Follow-up to #53. 
* This adds several escapes (HTML entity replacements) to the `escapeXML.pl` script from that PR

Separated out for a subsequent PR:
* (Re-)Running the improved `escapeXML.pl` (which processes all `.xml` files in the `import` folder)
* Such that `xmllint --noout import/*.xml` returns successfully (reports no errors)

danielgildea commented:
We should really fix the script that generates the xml files in the first place: https://github.com/acl-org/acl-pub/issues/9

CTNLP commented:
Hello @chbrown I just created I pull request (https://github.com/acl-org/acl-anthology/pull/75) for a version of the anthology xml files in which the <von> problem is fixed. Could you include the changes from that pull request in your pull request? If you also do the re-base to the current master for everything else, then I think we should be able to include your changes super fast. If you do not have time to do this then that would be understandable. We will still try our best to include your changes.

chbrown commented:
Hi @CTNLP, I can rebase it off master on your fork, but I don't think GitHub will recognize that as a PR on this repo? But maybe? I can try, but it seems like it'd be easier (less work for me, less hassle for PR reviewers) just to rebase on @acl-org/acl-anthology:master _after_ you merge #75, which strikes me as a better separation of concerns anyway.

P.S. It won't be a straightforward rebase at this stage, anyway, since you affected some of the same files in #75 that this PR affects. But that was a bygone conclusion due to changes to these files on master since June, when I submitted #61, regardless of #75.

knmnyn commented:
Hi all:

I agree that we should merge #76 before trying to redo #75 via @chbrown.
This seems like the right thing to do.



On Mon, Aug 27, 2018 at 11:08 AM Christopher Brown <notifications@github.com>
wrote:

> Hi @CTNLP <https://github.com/CTNLP>, I can rebase it off master on your
> fork, but I don't think GitHub will recognize that as a PR on this repo?
> But maybe? I can try, but it seems like it'd be easier (less work for me,
> less hassle for PR reviewers) just to rebase on
> @acl-org/acl-anthology:master *after* you merge #75
> <https://github.com/acl-org/acl-anthology/pull/75>, which strikes me as a
> better separation of concerns anyway.
>
> P.S. It won't be a straightforward rebase at this stage, anyway, since you
> affected some of the same files in #75
> <https://github.com/acl-org/acl-anthology/pull/75> that this PR affects.
> But that was a bygone conclusion due to changes to these files on master
> since June, when I submitted #61
> <https://github.com/acl-org/acl-anthology/pull/61>, regardless of #75
> <https://github.com/acl-org/acl-anthology/pull/75>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/acl-org/acl-anthology/pull/61#issuecomment-416296333>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANP63CAwHcYYI2H_uQ5fitXFwFS5LlZks5uVCdzgaJpZM4UhE1F>
> .
>

chbrown commented:
Okay I split out the non-overlapping bits and updated this PR with just the parallelizable change (squashed into one commit, for simplicity). I can re-run the XML fixes and submit a new PR once #75 is merged.

CTNLP commented:
Great, thank you. @villalbamartin is currently testing the other PR. We will get to your PR right after.

villalbamartin commented:
I've re-run the import with this version of the script, and I'm afraid we still have problems with the '&' characters. That said, I think all of the new escaped entities should be in the script, and therefore I'll accept the PR now and move the discussion to an issue.
